### PR TITLE
Test on Python 3.12 and Django 5.0

### PR DESCRIPTION
**Pull request by adamchainz**
Based on the changes made in #235.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -9,7 +9,12 @@ jobs:
       fail-fast: false
       max-parallel: 5
       matrix:
-        python-version: ['3.8', '3.9', '3.10', '3.11']
+        python-version:
+        - '3.8'
+        - '3.9'
+        - '3.10'
+        - '3.11'
+        - '3.12'
```
Comment (adamchainz) [inline .github/workflows/test.yml:17]: Splitting one per line reduces future diff noise when adding/removing versions

File: Makefile [hunk 2]
```diff
@@ -22,7 +22,7 @@ docs:
 test:
 	@flake8
 	@isort --check-only --diff formtools tests
-	@ coverage run `which django-admin` test tests
+	@ python -W error::DeprecationWarning -W error::PendingDeprecationWarning -m coverage run `which django-admin` test tests
```
Comment (adamchainz) [inline Makefile:25]: I use this pattern on all my projects to expose all warnings - even those from coverage or Django.

File: tests/requirements.txt [hunk 3]
```diff
@@ -1,3 +1,3 @@
-coverage==4.5.4
+coverage==7.3.2
```
Comment (adamchainz) [inline tests/requirements.txt:1]: Upgrade needed for 3.12 compatibility

File: tox.ini [hunk 4]
```diff
@@ -1,9 +1,10 @@
 [tox]
 args_are_paths = false
 envlist =
-    py{37,38,39,310,311}-django32
+    py{38,39,310,311}-django32
```
Comment (adamchainz) [inline tox.ini:4]: dropping 37 missed in 1d19b68

**Review by claudep — APPROVED** [review]
Thanks a lot! It always help to have an Adam in your pocket!

**Comment by adamchainz** [comment]
Woops, looks like Python 3.11 is actually broken. I'll take a look into this tomorrow.

**Comment by claudep** [comment]
Ah yes, I initially thought it was a CI issue, but then realized that Python 3.11 run is also launching 3.2 and 4.0 tests, and it shouldn't.

**Pull request merged**